kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,463 @@
+        }
+        // if namespace is not set we install operator to 'kroxylicious-operator'
+        if (this.namespaceInstallTo == null || this.namespaceInstallTo.isEmpty()) {
+            this.namespaceInstallTo = Constants.KO_NAMESPACE;
+        }
+        if (this.namespaceToWatch == null) {
+            this.namespaceToWatch = this.namespaceInstallTo;
+        }
+        if (this.bindingsNamespaces == null) {
+            this.bindingsNamespaces = new ArrayList<>();
+            this.bindingsNamespaces.add(this.namespaceInstallTo);
+        }
+        if (this.operationTimeout == null) {
+            this.operationTimeout = Constants.KO_OPERATION_TIMEOUT_DEFAULT;
+        }
+        if (this.reconciliationInterval == null) {
+            this.reconciliationInterval = Constants.RECONCILIATION_INTERVAL;
+        }
+        if (this.extraLabels == null) {
+            this.extraLabels = new HashMap<>();
+        }
+    }
+
+    private static List<File> getOperatorFiles() {
+        if (operatorFiles == null || operatorFiles.isEmpty()) {
+            operatorFiles = Arrays.stream(Objects.requireNonNull(new File(Constants.PATH_TO_OPERATOR_INSTALL_FILES).listFiles())).sorted()
+                    .filter(File::isFile)
+                    .toList();
+        }
+        return operatorFiles;
+    }
+
+    private static List<File> getFilteredOperatorFiles(Predicate<File> predicate) {
+        return getOperatorFiles().stream().filter(predicate).toList();
+    }
+
+    private boolean isKroxyliciousOperatorNamespaceNotCreated() {
+        return extensionContext == null
+                || extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.PREPARE_OPERATOR_ENV_KEY + namespaceInstallTo) == null;
+    }
+
+    private void createKroxyliciousOperatorNamespaceIfPossible() {
+        if (isKroxyliciousOperatorNamespaceNotCreated()) {
+            cluster.setNamespace(namespaceInstallTo);
+
+            if (this.extensionContext != null) {
+                ResourceManager.getStoredResources().computeIfAbsent(this.extensionContext.getDisplayName(), k -> new Stack<>());
+                ResourceManager.getStoredResources().get(this.extensionContext.getDisplayName()).push(
+                        new ResourceItem<>(this::deleteKroxyliciousOperatorNamespace));
+
+                NamespaceUtils.createNamespaces(namespaceInstallTo, bindingsNamespaces);
+
+                this.extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).put(Constants.PREPARE_OPERATOR_ENV_KEY + namespaceInstallTo, true);
+            }
+        }
+    }
+
+    /**
+     * Perform application of ServiceAccount, Roles and CRDs needed for proper cluster operator deployment.
+     * Configuration files are loaded from kroxylicious-operator directory.
+     */
+    public void applyClusterOperatorInstallFiles(String namespaceName) {
+        for (File operatorFile : getFilteredOperatorFiles(installFiles)) {
```

Comment:
I think you should sort the filenames so that namespace (00) is processed first.

Reply:
It is sorted. It is done in the private method `getOperatorFiles()`

Reply:
It is worthwhile making that clear in the method name  - maybe `getOrderedOperatorInstallFiles`